fix(iHeartRadio): fixed song and station presence/data

iiTxicz opened the pull request:
## Description 
<!-- A clear and detailed description of the changes, referencing issues if applicable -->

- Fixes issue #8150 

## Acknowledgements
- [✅ ] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [ ] I linted the code by running `yarn format`
- [ ✅] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots
<details>
<summary> Proof showing the creation/modification is working as expected </summary>
<!-- 
    Screenshots of the presence settings (if applicable) and at least TWO screenshots of the presence displaying correctly
    Including these screenshots will assist the reviewing processes thus speeding up the process of the pull request being merged
-->

https://cdn.rrc.bz/static/jackasu/2024-06-13+20-45-09.mp4
https://cdn.rrc.bz/static/jackasu/proof0.PNG
https://cdn.rrc.bz/static/jackasu/proof1.PNG

</details>

Bas950 reviewed (changes requested):
Fix the errors from compile lint etc.

iiTxicz commented:
everything should be good to go, had a quick question. Is it possible to get added a contributor? Fixed a good amount of bugs and added some new features to the presence.

iiTxicz commented:
Also, had another question for another presence I'm trying to fix, do you guys accept wildcard domains? I'm trying to revise the edgenuity presence but i'm not able to do *.core.learn.edgenuity.com cause its not part of the regex, and since I'm not for sure how many domains are ending in r*, it'll be a pain to try and find all when i believe there are 20+ regions. Ex. r03.core.learn.edgenuity.com

DarkVillager commented:
> Also, had another question for another presence I'm trying to fix, do you guys accept wildcard domains? I'm trying to revise the edgenuity presence but i'm not able to do `*`.core.learn.edgenuity.com cause its not part of the regex, and I don't know how many subdomains are ending in r`*`, think there are around 20+ regions. Ex. r03.core.learn.edgenuity.com

uhhh just add it to regex?

DarkVillager commented:
> everything should be good to go, had a quick question. Is it possible to get added as a contributor to the iHeartRadio presence? Fixed a good amount of bugs and added some new features to the presence.

Imo yes.

DarkVillager commented on websites/I/iHeartRadio/presence.ts:
```diff
@@ -75,16 +107,63 @@ presence.on("UpdateData", async () => {
+					: null;
+
+				presenceData.largeImageKey = srcImage.includes("ops=fit")
+					? `${srcImage}`
```
```suggestion
					? srcImage
```

DarkVillager commented on websites/I/iHeartRadio/presence.ts:
```diff
@@ -46,10 +29,29 @@ presence.on("UpdateData", async () => {
+						: null;
+
+					presenceData.largeImageKey = srcImage.includes("ops=fit")
+						? `${srcImage}`
```
```suggestion
						? srcImage
```

iiTxicz commented:
Everything should be good to go now, feel free to let me know if anything else is needed. If not, feel free to merge whenever, appreciate it.

theusaf commented on websites/I/iHeartRadio/presence.ts:
```diff
@@ -60,12 +62,42 @@ presence.on("UpdateData", async () => {
 				subtitle = checkLength(subtitle);
 				presenceData.state = subtitle;
 
+				if (document.URL.includes("live")) {
+					const stationData = (presenceData.buttons = [
+						{
+							label: "View Station",
+							url: document.URL,
```
Use `document.location.href` instead of `document.URL`

theusaf commented on websites/I/iHeartRadio/presence.ts:
```diff
@@ -75,16 +107,67 @@ presence.on("UpdateData", async () => {
+					?.querySelector('img[alt="Player Artwork Image"]')
+					?.getAttribute("src");
+
+				srcImage
+					? (srcImage = srcImage.replace(
+							"ops=fit(150%2C150)",
+							"ops=fit(512%2C512)"
+					  ))
+					: null;
```
Use `if` statements rather than ternaries for better readability.

theusaf commented on websites/I/iHeartRadio/presence.ts:
```diff
@@ -60,12 +62,42 @@ presence.on("UpdateData", async () => {
+							}
+						}
+
+						if (links.length > 0) {
+							links[1].includes("songs")
+								? stationData.push({
+										label: "View Song",
+										url: links[1],
+								  })
```
combine this with the `if` statement.

theusaf commented on websites/I/iHeartRadio/presence.ts:
```diff
@@ -94,18 +177,20 @@ presence.on("UpdateData", async () => {
 				(presenceData.smallImageText = paused
 					? (await strings).pause
 					: (await strings).play),
-				([presenceData.startTimestamp, presenceData.endTimestamp] =
-					presence.getTimestamps(parsedTimestamps[0], parsedTimestamps[1]));
+				([presenceData.startTimestamp, presenceData.endTimestamp] = [
+					timestamps[0],
+					timestamps[1],
+				]);
```
Separate all assignment like these with `;` rather than `,`

iiTxicz commented:
Everything should be good to go now
